### DISCUSSION TITLE
Thoughts and observations, mostly regarding wiki search results

### DISCUSSION THREAD

**Discussion by lemtom** [MarginaliaSearch/MarginaliaSearch#115] [Ideas]
## Independent wikis
I've noticed a preference for All The Tropes over TVTropes in search results.

In a similar vein, would it make sense to prefer the independent wikis listed by [Indie Wiki Buddy](https://getindie.wiki/listings/) over their (commercial) Fandom alternatives? I also noticed that not all of them are indexed yet, but I fear mass-submitting these might come off as spammy.

## BreezeWiki
BreezeWiki is a service that removes ads and clunk from Fandom wiki pages (also supported by the aforementioned Indie Wiki Buddy add-on). Links to articles cleaned by Breezewiki could perhaps be interpreted as links to the respective Fandom wikis, for the backlinking to be more accurate, but I understand that this could be too transformative with regards search results.

### Expanded scope
If transforming the search results in any way is in scope, some more potential avenues:
* Opening Fandom results with BreezeWiki, or offering alternative Wikipedia front-ends to open Wikipedia results (e.g. encyclopedia.marginalia.nu)
* Coupling Indie and Fandom wiki results per the previous section, since in many cases they duplicate content. 

This could probably also just be browser add-on behavior, though.

## Reddit
Reddit results tend to show up when I filter for Wiki results (specifically Old Reddit). Is that intended behavior? Relatedly, should Old Reddit, np.reddit.com and regular Reddit truly be considered different sites?

**Comment by vlofgren** [comment]
Hmm, all of these ideas sound good.  Will definitely import all the URLs in the indie wiki list you linked to ensure they're crawled :-)

Will test to see how feasible translating fandom wiki URLs to breezewiki URLs would be.  I agree that fandom is a bit of a nuisance with its aggressive ads.

Will also look into the reddit thing, that seems like a bug. 

If you have any other ideas for domains to promote or use as subsitutes, I'd love to hear them.

**Reply by lemtom** [reply]
Ah, that's wonderful. It's truly been a delight to see this magnificent project grow.

For Wikis specifically, Indie Wiki Buddy probably has the most curated one-to-one list. Back in the day I was tangentially involved in several moves away from Fandom (then Wikia), and most such moves were documented by the [Anti-Wikia Alliance](https://awa.shoutwiki.com/wiki/Forked_wikis). But from skimming the list, nothing immediately pops out as missing from Indie Wiki Buddy.

I use add-ons to make Medium behave, but [Scribe](https://scribe.rip) and Libmedium(https://libmedium.batsense.net) seem to work fine. [AnonymousOverflow](https://aohub.httpjames.space/) cleans up StackOverflow and StackExchange pages, because StackOverflow is kinda annoying to browse when not logged in. I do miss the days of alternative frontends for Reddit and Twitter, but they've cracked down on those.

Stray thought: Should archive.is links also be "unwrapped" to be clear backlinks-wise what site is actually being linked to?

## Catholic Encyclopedia
Some random thing that popped up in my head, since I used this public domain encyclopedia extensively during my own studies.

New Advent pops up first when I look for the Catholic Encyclopedia, but it's a very ad-heavy site with arbitrarily added hyperlinks. There are several other sites that offer this encyclopedia ([CatholiCity, for example](https://www.catholicity.com/encyclopedia/)), and there's also the one on [Wikisource](https://en.wikisource.org/wiki/Catholic_Encyclopedia_(1913)). Wikisource is already crawled, and as far as it concerns the Catholic Encyclopedia parts of both New Advent and Wikisource, the actual text is identical. Presumably, all the other public domain texts on New Advent are also on Wikisource, but that's probably less of a direct concern.

Looking into it further, the [search results for Catholic Encyclopedia](https://search.marginalia.nu/search?query=catholic+encyclopedia) as a whole are kinda odd. I would've expected the English Wikipedia article about it popping up, but instead I see a search result of the mobile version of the Italian Wikipedia, and then later a template of the mobile version of the Simple English Wikipedia. And then later there's a template from … whatever "en.wikipedia.beta.wmflabs.org" is. Maybe wiki templates (articles that start with "Template:") should be excluded from search results, and the mobile versions should probably be considered the same as the non-mobile versions. Since non-English results are not in scope right now, my thought is that the Italian result crept in because the title is English? That seems to be the general vibe of [what is indexed for the domain](https://search.marginalia.nu/site/it.m.wikipedia.org?view=docs).

## Other odd results
Some other results I noticed while trying some queries. In the future: should I report these with the report feature, or make Github issues out of them?

* [A search for a trading card](https://search.marginalia.nu//search?query=%22blue-eyes+white+dragon%22&profile=wiki&js=default&adtech=default&recent=default&searchTitle=default&newfilter=true) gives me a Vietnamese Fandom wiki result, instead of its English counterpart. In fact, the [entire site](https://search.marginalia.nu/site/yugioh.fandom.com) seems to be indexed as its Vietnamese version. Probably in part related to the Vietnamese language wiki having a lot of English language articles for some reason.

* When [searching for Belgian comic series Suske and Wiske](https://search.marginalia.nu/search?query=suske+and+wiske), I stumbled upon [a domain that appears to be a machine-translated version of the German wikipedia](https://search.marginalia.nu/site/de.zxc.wiki). The sample URLs are corrupted, it seems. The search also retrieved several Dutch language results, is that considered a problem at this stage? (In fact I encounter a lot of non-English language results during my searches)

**Comment by vlofgren** [comment]
These have been great contributions!

I've fixed the bug that has reddit content showing up as wiki stuff, but it'll be a few weeks before that's visible as I need to rebuild the index.

I've also added transparent translation to breezewiki and scribe.rip for fandom and medium respectively.  Holy crap that's an improvement :D 

There's still more stuff to do in this domain, but I'm a bit occupied with wrapping up another huge change so I've just picked the easiest stuff for now.  Will revisit this when I have more time.

**Reply by lemtom** [reply]
Very exciting!

Just a heads-up: the Breezewiki transformation breaks the Vietnamese Yu-Gi-Oh results. Seems like Breezewiki only supports English language Fandom wikis for now? Since the Vietnamese inclusion is probably an anomaly, it doesn't directly matter right now, but that's probably something to keep in mind when expanding the scope of Marginalia Search to other languages in the future.

**Reply by vlofgren** [reply]
Yup, I spotted that I think as you were commenting, should already be fixed.  I just dropped the language prefix entirely for now, as the search engine doesn't officially support other languages.
